### PR TITLE
ECO-1579 Autocreation of abstract product

### PR DESCRIPTION
**Pull request by vhrychenko**

### CODE REVIEW

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Stream/DataImport/DataImportProductConcreteWriteStream.php [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+        return true;
+    }
+
+    /**
+     * @param string $sku
+     *
+     * @return string
+     */
+    protected function createAbstractSKU(string $sku): string
```
Comment (valerio8787) [inline src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Stream/DataImport/DataImportProductConcreteWriteStream.php]: I think this creation should be in translator

Comment (valerio8787) [reply src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Stream/DataImport/DataImportProductConcreteWriteStream.php]: Generate Abstract sku in Translator also add mark that abstract Product should be created inside translator

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Mapper/Map/ProductImportMap.php [hunk 2]
```diff
@@ -7,20 +7,68 @@
...
-            'prices' => 'values.price',
+            'prices' => function ($item) {
+                $result = [];
+                foreach ($item['values']['price'] as $value) {
+                    $result[] = [
+                        'price' => $value['price'],
+                        'price_type' => $value['type'],
+                        'currency' => $value['currency'],
+                        'store' => $value['store'],
+//                        'abstract_sku' => $item['identifier'],
```
Comment (valerio8787) [inline src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Mapper/Map/ProductImportMap.php]: Please remove comments

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Translator/TranslatorFunction/TranslatorFunctionFactory.php [hunk 3]
```diff
@@ -33,6 +41,14 @@ public function createAddAttributeOptionsTranslatorFunction(): TranslatorFunctio
         return new AddAttributeOptions($this->akeneoPimService);
     }
 
+    /**
+     * @return \SprykerMiddleware\Zed\Process\Business\Translator\TranslatorFunction\TranslatorFunctionInterface
+     */
+    public function createAddUrlToLocalizedAttributesTranslatorFunction(): TranslatorFunctionInterface
+    {
+        return new AddUrlToLocalizedAttributes($this->facade);
```
Comment (valerio8787) [inline src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Translator/TranslatorFunction/TranslatorFunctionFactory.php]: Can you create UrlStrategy here and pass it to translator instead facade?

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Mapper/Map/ProductImportMap.php (hunk 2 above)
Comment (vhrychenko) [reply src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Mapper/Map/ProductImportMap.php]: Done

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Translator/TranslatorFunction/TranslatorFunctionFactory.php (hunk 3 above)
Comment (vhrychenko) [reply src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Translator/TranslatorFunction/TranslatorFunctionFactory.php]: Done

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Stream/DataImport/DataImportProductConcreteWriteStream.php (hunk 1 above)
Comment (vhrychenko) [reply src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Stream/DataImport/DataImportProductConcreteWriteStream.php]: Done

File: src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Translator/Generator/UrlGeneratorStrategy.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+     *
+     * @return string
+     */
+    protected function generateUrlSlug($value): string
+    {
+        if (function_exists('iconv')) {
+            $value = iconv('UTF-8', 'ASCII//TRANSLIT', $value);
+        }
+
+        $value = preg_replace("/[^a-zA-Z0-9 -]/", "", trim($value));
```
[the remark this answers is not in the document]
Comment (vhrychenko) [reply src/SprykerEco/Zed/AkeneoPimMiddlewareConnector/Business/Translator/Generator/UrlGeneratorStrategy.php]: Done

**Pull request merged**